packages/kbn-config-schema/src/internals/index.ts

```diff
@@ -250,22 +250,47 @@ export const internals = Joi.extend([
 
     base: Joi.object(),
     coerce(value: any, state: State, options: ValidationOptions) {
-      // If value isn't defined, let Joi handle default value if it's defined.
-      if (value !== undefined && !isPlainObject(value)) {
-        return this.createError('object.base', { value }, state, options);
+      if (value === undefined || isPlainObject(value)) {
+        return value;
       }
 
-      return value;
+      if (options.convert && typeof value === 'string') {
+        try {
+          const parsed = JSON.parse(value);
+          if (isPlainObject(parsed)) {
+            return parsed;
+          }
+          return this.createError('object.base', { value: parsed }, state, options);
+        } catch (e) {
+          return this.createError('object.parse', { value }, state, options);
+        }
+      }
+
+      return this.createError('object.base', { value }, state, options);
     },
     rules: [anyCustomRule],
   },
   {
     name: 'map',
 
     coerce(value: any, state: State, options: ValidationOptions) {
+      if (value === undefined) {
+        return value;
+      }
       if (isPlainObject(value)) {
         return new Map(Object.entries(value));
       }
+      if (options.convert && typeof value === 'string') {
```

Comment:
Note: I'm checking `options.convert` here to see if we should try to convert the value, However this option is never set in our library, and will always default to the default joi value which is `true`. It was more like a futur-proof thing if we want to implement this option at some point.